### PR TITLE
Lambda Layer Update- Update OTel JS to 1.6

### PR DESCRIPTION
**Pull request by Kausik-A**
This PR updates OTel JS to `1.6`, OTel API to `1.2.0` and other sdk dependencies to the latest available.

### CODE REVIEW

File: nodejs/packages/layer/package.json [hunk 1]
```diff
@@ -26,29 +26,29 @@
     "node": ">=10.0.0"
```
Comment (Aneurysm9) [inline nodejs/packages/layer/package.json]: Should this be `>=12.0.0`?  v10 is not supported by lambda.

Comment (vasireddy99) [reply nodejs/packages/layer/package.json]: +1

Comment (Kausik-A) [reply nodejs/packages/layer/package.json]: updated

File: nodejs/packages/layer/package.json [hunk 2]
```diff
@@ -23,32 +23,32 @@
   "author": "OpenTelemetry Authors",
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=10.0.0"
+    "node": ">=12.0.0"
   },
   "dependencies": {
-    "@opentelemetry/api": "^1.1.0",
+    "@opentelemetry/api": "^1.2.0",
```
Comment (vasireddy99) [inline nodejs/packages/layer/package.json:29]: The latest tag is still pointed to 1.1.0. we should double check to use the `next` tag, maybe better to use the version that has `latest` tag.
image

**Comment by vasireddy99** [comment]
I think we can update all the dependencies to the latest available. 
eg. 

```
    "@opentelemetry/exporter-trace-otlp-proto": "^0.32.0",
    "@opentelemetry/exporter-metrics-otlp-proto": "^0.32.0",
```

File: nodejs/packages/layer/package.json (hunk 2 above)
Comment (Kausik-A) [reply nodejs/packages/layer/package.json:29]: The [opentelemetry-js-api](https://github.com/open-telemetry/opentelemetry-js-api/releases) shows `1.2.0` to be the `latest` and doesn't call out anywhere to be RC. Is there a reason to still be using `1.1.0` ?

File: nodejs/packages/layer/package.json [hunk 3]
```diff
@@ -23,32 +23,32 @@
...
     "@opentelemetry/exporter-trace-otlp-proto": "^0.31.0",
     "@opentelemetry/exporter-metrics-otlp-proto": "^0.31.0",
     "@opentelemetry/instrumentation": "^0.31.0",
     "@opentelemetry/instrumentation-aws-lambda": "^0.32.0",
-    "@opentelemetry/instrumentation-aws-sdk": "^0.8.0",
+    "@opentelemetry/instrumentation-aws-sdk": "^0.8.1",
     "@opentelemetry/instrumentation-dns": "^0.29.0",
     "@opentelemetry/instrumentation-express": "^0.30.0",
     "@opentelemetry/instrumentation-graphql": "^0.29.0",
     "@opentelemetry/instrumentation-grpc": "^0.31.0",
```
Comment (vasireddy99) [inline nodejs/packages/layer/package.json]: "^0.32.0" is available for us to update 🙂

File: nodejs/packages/layer/package.json (hunk 2 above)
Comment (vasireddy99) [reply nodejs/packages/layer/package.json:29]: Yeah, not sure why it is not targeted to latest in `npm` though. 
image

Comment (vasireddy99) [reply nodejs/packages/layer/package.json:29]: Looks like it is safe to use the version `1.2.0`. it is compatible and can confirm from here : https://github.com/open-telemetry/opentelemetry-js/pull/3177

File: nodejs/packages/layer/package.json (hunk 3 above)
Comment (Kausik-A) [reply nodejs/packages/layer/package.json]: updated :)

**Review by vasireddy99 — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]

**Pull request merged**